source/common/network/listen_socket_impl.h

```diff
@@ -33,6 +33,35 @@ class ListenSocketImpl : public SocketImpl {
   void setupSocket(const Network::Socket::OptionsSharedPtr& options);
   void setListenSocketOptions(const Network::Socket::OptionsSharedPtr& options);
   Api::SysCallIntResult bind(Network::Address::InstanceConstSharedPtr address) override;
+
+  void close() override {
+    if (io_handle_ != nullptr) {
+      if (io_handle_->isOpen()) {
+        io_handle_->close();
+      }
+    } else {
+      std::stringstream out;
+      connection_info_provider_->dumpState(out, 0);
+      std::string state = out.str();
+      ENVOY_BUG(false, fmt::format("close() is called from ListenSocketImpl but the io handle is "
+                                   "nullptr. Info: {}",
+                                   out.str()));
```

Comment:
Not really a fan of putting this in here. Either it can happen or it can't. If it can happen just add comments and have the error checking? Same below.

Reply:
By design it should not happen. Are you saying we let it crash and dump via RELEASE_ASSERT?

Reply:
What I'm saying is you have the null checking above, so just remove this entire else branch, or just remove the null checking and fix the callsites?

Reply:
Thanks. Add the null check with RELEASE_ASSERT which can dump more logs.

The callsites is not an option because all the place I can think of doesn't hit that condition